### PR TITLE
Ensure static index signatures have an errorNode available

### LINKED ISSUE #44082: Static index signatures do not ensure consistency across index kinds within the same class
### ⏯ Playground Link

[Playground link](https://www.typescriptlang.org/play?ts=4.4.0-dev.20210513#code/MYGwhgzhAECK0G8BQ1XQE4FMwBMD2AdiAJ7QDaEAXNBAC7oCWBA5gLrUECuAtgEaboA3CjRZchEuSrQufAe2gAWAEzQAPtGUBmLUgC+SJKEgwA6tEwAPWpgI4Y8ZGgzZ8RUhWp1GLBbP7o+obGUNAAouZWNnZmiCKoYm6SnjT0TGzUvHh4INgE8S7i7lIcPAEKAIzQQcHgoQBCcc50YLQMwIVJHtLe6X5lAsLNtK3tnRLdpXLoCirqmjo1ITAAIhbWtvbQjU5oLW0diRMlqT4ZMgOBBkZ1MGFrUZurTXsjB+PFKb2+mdm5YPlhqNDq5jil-PJqFUgkA)

### 💻 Code

```ts-repro
// non-static equivalent, for reference
class Q {
    readonly [s: string]: number;
    readonly [s: number]: 42 | 233
}

class W extends Q {
    readonly [s: string]: number
}

class EW extends W {
    readonly [s: string]: boolean
    readonly [s: number]: 1 
}

// static example
class B {
    static readonly [s: string]: number;
    static readonly [s: number]: 42 | 233
}

class D extends B {
    static readonly [s: string]: number
}

class ED extends D {
    static readonly [s: string]: boolean
    static readonly [s: number]: 1 // should error, 1 isn't assignable to `boolean`!
}

```

### 🙁 Actual behavior

No error.

### 🙂 Expected behavior

`Numeric index type '1' is not assignable to string index type 'boolean'.`

### LINKED ISSUE #44134: Incorrect error span on classes with both instance and static index signatures
```ts
class Foo {
  [p: string]: any;
  static [p: string]: number;
}
```

```
error: Property 'prototype' of type 'Foo' is not assignable to string index type 'number'.
```

### PR DESCRIPTION
**Pull request by weswigham**
So they can actually report the compatibility error they were trying to issue when the string and number index signatures mismatched.

Fixes #44082
Fixes #44134

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -36898,6 +36898,11 @@ namespace ts {
                     const someBaseTypeHasBothIndexers = forEach(getBaseTypes(<InterfaceType>type), base => getIndexTypeOfType(base, IndexKind.String) && getIndexTypeOfType(base, IndexKind.Number));
                     errorNode = someBaseTypeHasBothIndexers || !type.symbol.declarations ? undefined : type.symbol.declarations[0];
                 }
+                if (!errorNode) {
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts:36901]: Seems like this isn't the right fix - you'll still end up with the wrong error span, just as in the following example:

```ts
class Foo {
  [p: string]: any;
  static [p: string]: number;
}
```

```
error: Property 'prototype' of type 'Foo' is not assignable to string index type 'number'.
```

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: src/compiler/checker.ts (hunk 1 above)
Comment (DanielRosenwasser) [reply src/compiler/checker.ts:36901]: I've opened up #44134 to track that.

**Review by DanielRosenwasser — APPROVED** [review]

**Comment by weswigham** [comment]
@DanielRosenwasser This now fixes both reporting issues and, per our discussion to mitigate the impact of that and preserve the usefulness of the feature, also disables checking if index signatures are consistent with prototype properties on class static sides.

**Comment by weswigham** [comment]
@DanielRosenwasser you want a cherry pick for this, I assume?

@typescript-bot cherry-pick this into release-4.3

**Pull request merged**